packages/nextjs/services/api/batches/index.ts

```diff
@@ -0,0 +1,96 @@
+import { SortingState } from "@tanstack/react-table";
+import { Batch, BatchWithCounts } from "~~/services/database/repositories/batches";
+
+export const getSortedBatches = async (start: number, size: number, sorting: SortingState, filter?: string) => {
+  const response = await fetch(
+    `/api/batches/sorted?start=${start}&size=${size}&sorting=${JSON.stringify(sorting)}&filter=${filter || ""}`,
+  );
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch sorted batches: ${response.status} ${response.statusText}`);
+  }
+
+  const batchesData = (await response.json()) as {
+    data: BatchWithCounts[];
+    meta: {
+      totalRowCount: number;
+    };
+  };
+
+  return batchesData;
+};
+
+export async function createBatch({
+  address,
+  signature,
+  name,
+  startDate,
+  status,
+  contractAddress,
+  telegramLink,
+  bgSubdomain,
+}: {
+  address: string;
+  signature: string;
+  name: string;
+  startDate: string;
+  status: string;
+  contractAddress?: string;
+  telegramLink: string;
+  bgSubdomain: string;
+}) {
```

Comment:
Ohh I think maybe you should take this type as `CreateBatchPayload` from `/api/batches/create/route.ts` similar for editBatch?

Reply:
thanks, updated [4300a41](https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/137/commits/4300a41a714d23457eadfa42ce411fedd1c5a77a)